src/services/chains/generic.chain.ts

```diff
@@ -0,0 +1,173 @@
+    }
+
+    get httpService(): HttpService {
+        return this._config.httpService;
+    }
+
+    get loggerService(): LoggerService {
+        return this._config.loggerService;
+    }
+
+    initialize = async (): Promise<LumClient> => {
+        // If we want to connect to RPC WS, we have to patch the connection URI
+        const useEndpoint = this._config.subscribeToRPC ? this._config.endpoint.replace('https://', 'wss://').replace('http://', 'ws://') : this._config.endpoint;
+
+        // Bind and acquire the chain id
+        this._client = await LumClient.connect(useEndpoint);
+        this._chainId = await this._client.getChainId();
+
+        // If we want to subscribe to RPC, we have to create a stream
+        if (this._config.subscribeToRPC) {
+            this._clientStream = this._client.tmClient.subscribeNewBlock();
+        }
+
+        // If we have a post init callback, just mention it
+        this.loggerService.debug(`Connected to ${this.symbol} chain = ${useEndpoint} (${this._chainId})`);
+        if (this._config.postInitCallback) {
+            this._config.postInitCallback(this);
+        }
+        return this._client;
+    };
+
+    isInitialized = (): boolean => {
+        return this._client !== undefined;
+    };
+
+    getTokenInformationFromOsmosis = async (): Promise<any> => {
+        try {
+            const response = await lastValueFrom(this.httpService.get(`${ApiUrl.GET_CHAIN_TOKENS}/${this.symbol}`).pipe(map((response) => response.data)));
+            return response;
+        } catch (error) {
+            return null;
+        }
+    };
+
+    getPrice = async (): Promise<number> => {
+        const infos = await this.getTokenInformationFromOsmosis();
+        if (!infos) {
+            return 0;
+        }
+        return infos.price;
+    };
+
+    getMarketCap = async (): Promise<number> => {
+        try {
+            const data = await lastValueFrom(this.httpService.get(ApiUrl.GET_CHAIN_TOKENS_MCAP).pipe(map((response) => response.data)));
+            const extractedData = data.find((d: any) => d.symbol === this.symbol);
+            return extractedData ? extractedData.market_cap : 0;
+        } catch (error) {
+            return 0;
+        }
+    };
+
+    getTokenSupply = async (): Promise<number> => {
+        try {
+            const supply = Number((await this.client.getSupply(this.microDenom)).amount) / TEN_EXPONENT_SIX;
+            return supply;
+        } catch (error) {
+            return 0;
+        }
+    };
+
+    getAPY = async (): Promise<number> => {
+        const inflation = Number(await this.client.queryClient.mint.inflation()) / CLIENT_PRECISION;
+        const metrics = await computeTotalApy(this.client, Number(await this.getTokenSupply()), inflation, CLIENT_PRECISION, TEN_EXPONENT_SIX);
+        return apy(metrics.inflation, metrics.communityTaxRate, metrics.stakingRatio);
+    };
+
+    getTotalAllocatedToken = async (): Promise<number> => {
```

Comment:
Should we follow the try catch principle as getTokenSupply and return 0 in the catch ?